### DISCUSSION TITLE
Training the model

### DISCUSSION THREAD

**Discussion by nirivas** [cnr-isti-vclab/TagLab#63] [Q&A] [answered]
Hey everyone,

I was able to delineate all the individuals of a species in one 10x10 plot. I exported the dataset and trained the network. How do I continue training the network using a different plot? This way it can learn from 2 or more plots and not just 1 before applying the automatic classification.

Apologies if this is an easy answer, couldn't find anything too clear on the manual

Thanks,

Nico

**Comment by maxcorsini** [comment, marked as answer]
Hi nirivas,

To train the network on two plots you have to do in the following way:

1) Export plot A (let's say in folder \mydata)
2) Export plot B choosing the same folder of the previous export (\mydata) . The data will be accumulated .
3) Train the network on such data (plotA + plotB) 

This is mandatory, if you train the network on the plot A, and then you train the network on the plot B, the network FORGET what has learned from the plot A. 

NOTE: You must use the same target scale factor. For example, plot A has a pixel size of 0.5mm and plot B of 0.7mm, a good choice for the target scale factor is 0.6 .


Cheers

**Reply by nirivas** [reply]
Fantastic! Thanks again!

**Comment by DrBuitragoLopez-SEZ** [comment]
Do you guys (developers) have any plan to provide a TagLab training course any time soon? I'd be very interested in learning how to get the best out of the machine learning algorithm. I'd like to get a proper automated "segmentation" and "classification", because the current manual method that I'm using is far from sustainable/efficient.

Cheers,

Carol

**Reply by damarislanders** [reply]
I would be very interested in this too :)

**Comment by Sharayujak** [comment]
Hi everyone,
I am done exporting my dataset and am trying to train the classifier now using "train your network function". after adding the dataset folder tag lab is unable to recognize classes. It says "an error occurred in your dataset, there is a mismatch between files, please export a new dataset". How do I troubleshoot this? 
Cheers,
Sharayu

**Comment by Sharayujak** [comment]
<img width="844" alt="mini" src="https://github.com/user-attachments/assets/99b1da3d-7e65-480d-b09f-4a1b501b2340">

**Comment by Sharayujak** [comment]
Hi everyone, I trained a classifier, and the results tab displayed a confusion matrix along with the validation loss and training loss graphs. However, I accidentally closed the window. Is there a way to access the confusion matrix again? Is it stored somewhere?
